samples/caf_sensor_manager/src/modules/sensor_sim_ctrl.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <drivers/sensor_sim.h>
+
+#include "sensor_sim_ctrl_def.h"
+
+#define MODULE sensor_sim_ctrl
+#include <caf/events/module_state_event.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE);
+
+enum state {
+	STATE_DISABLED,
+	STATE_ACTIVE,
+	STATE_ERROR
+};
+
+static enum state state;
+static uint8_t cur_wave_idx;
+static const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(sensor_sim));
+
+static void report_error(void)
+{
+	state = STATE_ERROR;
+	module_set_state(MODULE_STATE_ERROR);
+}
+
+static void set_wave(void)
```

Comment:
I think set_wave should be `static int set_wave()` and do not call `report_error()` inside, just return error code.